code/game/machinery/computer/Operating.dm

```diff
@@ -95,7 +94,27 @@
 		usr.set_machine(src)
 	return
 
-
 /obj/machinery/computer/operating/process()
 	if(..())
 		src.updateDialog()
+	update_icon()
+
+/obj/machinery/computer/operating/update_icon()
+	..()
+	icon_state = initial(icon_state)

```

Comment:
In the case that you set it equal to the SAME icon state. If so your else statement would have it redraw every tick despite having no change.